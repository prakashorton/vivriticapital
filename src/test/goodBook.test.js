import React from "react";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import GoodBook from "../components/goodBook";

configure({ adapter: new Adapter() });

describe("<Todo />", () => {
  let wrapper = null;

  beforeEach(() => {
    wrapper = mount(<GoodBook />);
  });

  it("render - snapshot check", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("render", () => {
    expect(wrapper).not.toBeNull();
  });

  it("render input box to search the displayed items", () => {
    const inputButton = wrapper.find("input");
    expect(inputButton.props().className).toBe("ant-input ant-input-lg");
    expect(inputButton.props().type).toBe("text");
    expect(inputButton.props().value).toBe("");
    expect(inputButton.props().placeholder).toBe(
      "Enter item to Search from [Brand - Variety - Style - Country - Stars]"
    );
  });
});
