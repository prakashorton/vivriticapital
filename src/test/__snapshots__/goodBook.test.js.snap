// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todo /> render - snapshot check 1`] = `
<GoodBook>
  <div
    className="container"
  >
    <div
      className="site-page-header-ghost-wrapper"
    >
      <PageHeader
        ghost={false}
        title="Search"
      >
        <div
          className="ant-page-header"
        >
          <div
            className="ant-page-header-heading"
          >
            <span
              className="ant-page-header-heading-title"
            >
              Search
            </span>
          </div>
          <div
            className="ant-page-header-content"
          >
            <Search
              enterButton={false}
              onChange={[Function]}
              placeholder="Enter item to Search from [Brand - Variety - Style - Country - Stars]"
              size="large"
            >
              <Input
                className="ant-input-search"
                onChange={[Function]}
                onPressEnter={[Function]}
                placeholder="Enter item to Search from [Brand - Variety - Style - Country - Stars]"
                prefixCls="ant-input"
                size="large"
                suffix={
                  <ForwardRef(SearchOutlined)
                    className="ant-input-search-icon"
                    onClick={[Function]}
                  />
                }
                type="text"
              >
                <ClearableLabeledInput
                  className="ant-input-search"
                  element={
                    <input
                      className="ant-input ant-input-lg ant-input-search"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      placeholder="Enter item to Search from [Brand - Variety - Style - Country - Stars]"
                      type="text"
                    />
                  }
                  focused={false}
                  handleReset={[Function]}
                  inputType="input"
                  onChange={[Function]}
                  onPressEnter={[Function]}
                  placeholder="Enter item to Search from [Brand - Variety - Style - Country - Stars]"
                  prefixCls="ant-input"
                  size="large"
                  suffix={
                    <ForwardRef(SearchOutlined)
                      className="ant-input-search-icon"
                      onClick={[Function]}
                    />
                  }
                  triggerFocus={[Function]}
                  type="text"
                  value=""
                >
                  <span
                    className="ant-input-search ant-input-affix-wrapper ant-input-affix-wrapper-lg"
                    onMouseUp={[Function]}
                  >
                    <input
                      className="ant-input ant-input-lg"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      placeholder="Enter item to Search from [Brand - Variety - Style - Country - Stars]"
                      style={null}
                      type="text"
                      value=""
                    />
                    <span
                      className="ant-input-suffix"
                    >
                      <ForwardRef(SearchOutlined)
                        className="ant-input-search-icon"
                        key="searchIcon"
                        onClick={[Function]}
                      >
                        <AntdIcon
                          className="ant-input-search-icon"
                          icon={
                            Object {
                              "icon": Object {
                                "attrs": Object {
                                  "focusable": "false",
                                  "viewBox": "64 64 896 896",
                                },
                                "children": Array [
                                  Object {
                                    "attrs": Object {
                                      "d": "M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z",
                                    },
                                    "tag": "path",
                                  },
                                ],
                                "tag": "svg",
                              },
                              "name": "search",
                              "theme": "outlined",
                            }
                          }
                          onClick={[Function]}
                        >
                          <span
                            aria-label="search"
                            className="anticon anticon-search ant-input-search-icon"
                            onClick={[Function]}
                            role="img"
                            tabIndex={-1}
                          >
                            <IconReact
                              className=""
                              icon={
                                Object {
                                  "icon": Object {
                                    "attrs": Object {
                                      "focusable": "false",
                                      "viewBox": "64 64 896 896",
                                    },
                                    "children": Array [
                                      Object {
                                        "attrs": Object {
                                          "d": "M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z",
                                        },
                                        "tag": "path",
                                      },
                                    ],
                                    "tag": "svg",
                                  },
                                  "name": "search",
                                  "theme": "outlined",
                                }
                              }
                            >
                              <svg
                                aria-hidden="true"
                                className=""
                                data-icon="search"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                key="svg-search"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                                  key="svg-search-svg-0"
                                />
                              </svg>
                            </IconReact>
                          </span>
                        </AntdIcon>
                      </ForwardRef(SearchOutlined)>
                    </span>
                  </span>
                </ClearableLabeledInput>
              </Input>
            </Search>
          </div>
        </div>
      </PageHeader>
    </div>
    <Empty>
      <LocaleReceiver
        componentName="Empty"
      >
        <div
          className="ant-empty"
        >
          <div
            className="ant-empty-image"
          >
            <Empty>
              <svg
                className="ant-empty-img-default"
                height="152"
                viewBox="0 0 184 152"
                width="184"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="none"
                  fillRule="evenodd"
                >
                  <g
                    transform="translate(24 31.67)"
                  >
                    <ellipse
                      className="ant-empty-img-default-ellipse"
                      cx="67.797"
                      cy="106.89"
                      rx="67.797"
                      ry="12.668"
                    />
                    <path
                      className="ant-empty-img-default-path-1"
                      d="M122.034 69.674L98.109 40.229c-1.148-1.386-2.826-2.225-4.593-2.225h-51.44c-1.766 0-3.444.839-4.592 2.225L13.56 69.674v15.383h108.475V69.674z"
                    />
                    <path
                      className="ant-empty-img-default-path-2"
                      d="M101.537 86.214L80.63 61.102c-1.001-1.207-2.507-1.867-4.048-1.867H31.724c-1.54 0-3.047.66-4.048 1.867L6.769 86.214v13.792h94.768V86.214z"
                      transform="translate(13.56)"
                    />
                    <path
                      className="ant-empty-img-default-path-3"
                      d="M33.83 0h67.933a4 4 0 0 1 4 4v93.344a4 4 0 0 1-4 4H33.83a4 4 0 0 1-4-4V4a4 4 0 0 1 4-4z"
                    />
                    <path
                      className="ant-empty-img-default-path-4"
                      d="M42.678 9.953h50.237a2 2 0 0 1 2 2V36.91a2 2 0 0 1-2 2H42.678a2 2 0 0 1-2-2V11.953a2 2 0 0 1 2-2zM42.94 49.767h49.713a2.262 2.262 0 1 1 0 4.524H42.94a2.262 2.262 0 0 1 0-4.524zM42.94 61.53h49.713a2.262 2.262 0 1 1 0 4.525H42.94a2.262 2.262 0 0 1 0-4.525zM121.813 105.032c-.775 3.071-3.497 5.36-6.735 5.36H20.515c-3.238 0-5.96-2.29-6.734-5.36a7.309 7.309 0 0 1-.222-1.79V69.675h26.318c2.907 0 5.25 2.448 5.25 5.42v.04c0 2.971 2.37 5.37 5.277 5.37h34.785c2.907 0 5.277-2.421 5.277-5.393V75.1c0-2.972 2.343-5.426 5.25-5.426h26.318v33.569c0 .617-.077 1.216-.221 1.789z"
                    />
                  </g>
                  <path
                    className="ant-empty-img-default-path-5"
                    d="M149.121 33.292l-6.83 2.65a1 1 0 0 1-1.317-1.23l1.937-6.207c-2.589-2.944-4.109-6.534-4.109-10.408C138.802 8.102 148.92 0 161.402 0 173.881 0 184 8.102 184 18.097c0 9.995-10.118 18.097-22.599 18.097-4.528 0-8.744-1.066-12.28-2.902z"
                  />
                  <g
                    className="ant-empty-img-default-g"
                    transform="translate(149.65 15.383)"
                  >
                    <ellipse
                      cx="20.654"
                      cy="3.167"
                      rx="2.849"
                      ry="2.815"
                    />
                    <path
                      d="M5.698 5.63H0L2.898.704zM9.259.704h4.985V5.63H9.259z"
                    />
                  </g>
                </g>
              </svg>
            </Empty>
          </div>
          <p
            className="ant-empty-description"
          >
            No Data
          </p>
        </div>
      </LocaleReceiver>
    </Empty>
  </div>
</GoodBook>
`;
